lib/Schedule/LongSteps/Storage/DynamoDB.pm

```diff
@@ -0,0 +1,538 @@
+    }
+
+    # Assume base64 encoded memGunzip
+    return JSON::from_json( Compress::Zlib::memGunzip( MIME::Base64::decode_base64( $dynamo_state ) ) );
+}
+
+sub _process_from_attrmap{
+    my ($self, $dynamoItem) = @_;
+    my $map = $dynamoItem->Map();
+
+    my $run_at = $map->{run_at}->S();
+    if( $run_at eq $TIME_MAX ){
+        $run_at = undef;
+    }else{
+        $run_at = DateTime::Format::ISO8601->parse_datetime( $run_at );
+    }
+    my $run_id = $map->{run_id}->S();
+    if( $run_id eq 'NULL' ){
+        $run_id = undef;
+    }
+    my $state = $self->_decode_state( $map->{mstate}->S() );
+    my $error = $map->{merror}->S();
+    if( $error eq 'NULL' ){
+        $error = undef;
+    }
+
+    return Schedule::LongSteps::Storage::DynamoDB::Process->new({
+        storage => $self,
+        id => $map->{id}->S(),
+        process_class => $map->{process_class}->S(),
+        status => $map->{mstatus}->S(),
+        what => $map->{what}->S(),
+        run_at => $run_at,
+        run_id => $run_id,
+        state => $state,
+        error => $error,
+    });
+}
+
+
+=head2 destroy_table
+
+Destroys this table. Mainly so tests don't leave some crap behind.
+
+Use that responsibly. Which is never except in tests.
+
+Note that this blocks until the table has effectively been deleted remotely.
+
+=cut
+
+sub destroy_table{
+    my ($self, $am_i_sure) = @_;
+    unless( $am_i_sure eq 'I am very sure and I am not insane' ){
+        confess("Sorry we cannot let you do that");
+    }
+    unless( $self->table_active() ){
+        confess("Sorry this table is not ACTIVE. Too early to destroy");
+    }
+    my $table_name = $self->table_name();
+    unless( $table_name =~ /^testdelete/ ){
+        confess("Sorry this is not a test table (from the test suite. Destroy manually if you are sure");
+    }
+    $log->warn("Will destroy $table_name from dynamoDB");
+    my $deletion = $self->dynamo_db()->DeleteTable(TableName => $table_name);
+    # Wait until the table is effectively destroyed.
+    while( $self->table_exists() ){
+        $log->warn("Table $table_name not destroyed yet. Waiting ".$self->creation_wait_time()." second");
+        if( $self->creation_wait_time() ){
+            sleep( $self->creation_wait_time() );
+        }
+    }
+    $log->warn("Table $table_name DESTROYED");
+    return $deletion;
+}
+
+__PACKAGE__->meta()->make_immutable();
+
+package Schedule::LongSteps::Storage::DynamoDB::Process;
+
+use Moose;
+
+use Log::Any qw/$log/;
+use JSON;
+use MIME::Base64;
+use Compress::Zlib;
+use DateTime;
+use DateTime::Format::ISO8601;
+use Data::Dumper;
+
+has 'storage' => ( is => 'ro', isa => 'Schedule::LongSteps::Storage::DynamoDB', required => 1 );
+has 'id' =>      ( is => 'ro', isa => 'Str', required => 1 );
+
+has 'process_class' => ( is => 'rw', isa => 'Str', required => 1); # rw only for test. Should not changed ever.
+has 'status' =>        ( is => 'rw', isa => 'Str', default => 'pending' );
```

Comment:
This seems a little messy, when you have taken the time to align the "is  => 'rw'"